common/src/main/java/com/microsoft/identity/common/internal/net/UrlConnectionHttpClient.java

```diff
@@ -22,325 +22,186 @@
 
-import static com.microsoft.identity.common.adal.internal.AuthenticationConstants.AAD.CLIENT_REQUEST_ID;
-import static com.microsoft.identity.common.internal.net.HttpUrlConnectionFactory.createHttpURLConnection;
+import static com.microsoft.identity.common.java.net.UrlConnectionHttpClient.DEFAULT_CONNECT_TIME_OUT_MS;
+import static com.microsoft.identity.common.java.net.UrlConnectionHttpClient.DEFAULT_READ_TIME_OUT_MS;
+import static com.microsoft.identity.common.java.net.UrlConnectionHttpClient.DEFAULT_STREAM_BUFFER_SIZE;
 
 /**
- * A client object for handling HTTP requests and responses.  This class accepts a RetryPolicy that
- * is applied to the responses.  By default, the policy is a null policy not retrying anything.  The
- * default settings for read timeout and connection timeout are 30s, and the default setting for the
- * size of the buffer for reading objects from the http stream is 1024 bytes.
- *
- * There are two ways to supply timeout values to this class, one method takes suppliers, the other
- * one integers.  If you use both of these, the suppliers method will take precendence over the method
- * using integers.
+ * Deprecated
+ * <p>
+ * Currently served as an adapter of {@link com.microsoft.identity.common.java.net.UrlConnectionHttpClient}
  */
-@AllArgsConstructor
-@Builder
-@ThreadSafe
-@Immutable
 public class UrlConnectionHttpClient extends AbstractHttpClient {
+
     /**
      * A functional interface modeled off of java.util.function.Supplier for providing
      * values to callers.
+     *
      * @param <T> the type of value provided.
      */
-    interface Supplier<T> {
-        /**
-         * @return an instance of a T.
-         */
-        T get();
+    public interface Supplier<T> extends com.microsoft.identity.common.java.util.ported.Supplier<T> {
     }
 
-    /**
-     * A utility method for creating suppliers of particular values.
-     * @param value the value to supply.
-     * @param <T> the type of the value to return.
-     * @return a new Supplier that returns the passed value.
-     */
-    public static <T> Supplier<T> supplierOf(final T value) {
-        return new Supplier<T>() {
-            public T get() {
-                return value;
-            }
-        };
+    private static transient AtomicReference<UrlConnectionHttpClient> defaultReference = new AtomicReference<>(null);
+
+    public static Builder builder() {
+        return new Builder();
     }
 
-    @Builder.Default
-    private final RetryPolicy<HttpResponse> retryPolicy = new NoRetryPolicy();
-    @Builder.Default
-    private final int connectTimeoutMs = 30000;
-    @Builder.Default
-    private final int readTimeoutMs = 30000;
-    @Builder.Default
-    private final Supplier<Integer> connectTimeoutMsSupplier = null;
-    @Builder.Default
-    private final Supplier<Integer> readTimeoutMsSupplier = null;
-    @Builder.Default
-    private final int streamBufferSize = 1024;
+    public static class Builder {
```

Comment:
we could probably use lombok here

Reply:
We can't. Since the constructor of this `adapter` actually initializes common-java's UrlConnectionHttpClient.